File: android/src/main/java/com/visioncameramlkit/domain/models/ImagePreprocessingOptions.kt
package com.visioncameramlkit.domain.models

data class ImagePreprocessingOptions(
  val invertColors: Boolean = false,
  // Android handles rotation automatically, so outputOrientation is ignored. Added for iOS parity.
  val outputOrientation: OutputOrientation? = null,
  val scaleFactor: Float = 1.0f,
  val orientation: Orientation = Orientation.PORTRAIT,
)

File: ios/Domain/Models/ImagePreprocessingOptions.swift
import Foundation

public struct ImagePreprocessingOptions {
  let invertColors: Bool
  let outputOrientation: OutputOrientation?
  let scaleFactor: CGFloat
  let orientation: DomainOrientation?
  let roi: DomainRegionOfInterest?

  init(
    invertColors: Bool = false,
    outputOrientation: OutputOrientation? = nil,
    scaleFactor: CGFloat = 1.0,
    orientation: DomainOrientation? = nil,
    roi: DomainRegionOfInterest? = nil
  ) {
    self.invertColors = invertColors
    self.outputOrientation = outputOrientation
    self.scaleFactor = scaleFactor
    self.orientation = orientation
    self.roi = roi
  }
}
